src/exercises/course_representation_types.py

```diff
@@ -0,0 +1,99 @@
+from collections import namedtuple
+from pathlib import Path
+from typing import List
+
+
+class Exercice(namedtuple("Exercise", ["name"])):
+    pass
+
+
+class Directory(namedtuple("Directory", ["name", "children"])):
+    def list_exercises(self, prefix) -> List[Path]:
+        res = []
+
+        dir_prefix = f"{self.name}/"
+
+        for child in self.children:
+            if isinstance(child, Exercice):
+                res.append(Path(f"{prefix}{dir_prefix}{child.name}.cairo"))
```

Comment:
The thing is that they don't have to exist at all time.
This can be initialized to then create the files afterward.
It's a sub tree of a file system, that can be append to any `parent` directory.

```
exercises = Directory(exercises_directory, course).list_exercises("")

current_working_exercises = Directory(
    current_working_exercises_directory, course
).list_exercises("")
```

We have a step that make sure exercises exist later on. It's in `src/exercises/list_test.py`

Reply:
oh you're right. So let's just stick to the `Path` API without verifying for existence, maybe you can convert all `prefix` you get in methods from `str` to `Path`? The prefix should and will actually exist in the filesystem